### ISSUE TITLE
There are some rar compression packages that do not throw an exception if no password is entered

### ISSUE THREAD

**Issue by yousa121380** [adamhathcock/sharpcompress#743]
Unzipping the following package using the example in the USAGE.md will throws an NullReferenceException
![image](https://github.com/adamhathcock/sharpcompress/assets/48246615/24369998-a51d-40c7-95f2-a759acf3f859)
 This is a rar package I downloaded from the Internet
 https://dz.certdon.com/down/2022down/3/18/NatTypeTesterrj.rar

**Comment by yousa121380** [comment]
mabe this is same issue 
https://github.com/adamhathcock/sharpcompress/issues/514#issuecomment-818490327

**Comment by Erior** [comment]
The link does not work for me ERR_NAME_NOT_RESOLVED

**Comment by yousa121380** [comment]
> The link does not work for me ERR_NAME_NOT_RESOLVED

After work I pull a request and put it in testcase

**Comment by yousa121380** [comment]
https://github.com/adamhathcock/sharpcompress/pull/744 i upload and add it@Erior

**Comment by Erior** [comment]
@yousa121380 , I created a pull where we now can skip over the encrypted files instead of throwing an exception, I guess that is what you were after?

**Comment by yousa121380** [comment]
> @yousa121380 , I created a pull where we now can skip over the encrypted files instead of throwing an exception, I guess that is what you were after?

yes, i test encrypted zip and it can throw exception then skip over, i want that it have same behave in  all encrypted files
